storage/ndb/rest-server/data-access-rondb/src/db-operations/pk/common.cpp

```diff
@@ -324,19 +329,17 @@ RS_Status SetOperationPKCol(const NdbDictionary::Column *col, NdbOperation *oper
   case NdbDictionary::Column::Char: {
     ///< Len. A fixed array of 1-byte chars
 
-    const int len = request->PKValueLen(colIdx);
-    if (len > col->getLength()) {
+    const int dataLen = request->PKValueLen(colIdx);
+    if (dataLen > col->getLength()) {
       return RS_CLIENT_ERROR(
           std::string(ERROR_008) +
-          " Data len is greater than column length. Column: " + std::string(col->getName()));
+          " Data length is greater than column length. Column: " + std::string(col->getName()));
     }
 
-    const char *charStr = request->PKValueCStr(colIdx);
-    char pk[col->getLength()];
-    for (int i = 0; i < col->getLength(); i++) {
-      pk[i] = 0;
-    }
-    memcpy(pk, charStr, len);
+    const char *charStr = request->PKValueCStr(colIdx);    
+    // operation->equal expects a zero-padded char string
+    char pk[MAX_KEY_SIZE_IN_WORDS*4] = { 0 };
```

Comment:
This is very nice, I did not know about this cpp extension. 
BTW it would be optimal to use `std.fill` to set only the elements according to the size of the primary key. But this can be done later.